### PR TITLE
Fixed single outFile generation (#1)

### PR DESCRIPTION
**Pull request by DomBlack**
- All input files now point to the same output file
- Added `AllowJS` compiler flag, as this is useful for projects using outFile
- Added `typingsFile` option to allow typings files to be written for older JS code being included in the single file output

### CODE REVIEW

File: src/main/resources/typescriptc.js [hunk 1]
```diff
@@ -266,6 +294,17 @@
             };
             output.results.push(result);
         });
+
+        // If we are outputting a single file create the array of written files
+        //  and map it against all input files
+        if (options.out) {
+            var filesWritten = createFilesWrittenArray(options.out);
```
Comment (BrandonArp) [inline src/main/resources/typescriptc.js:301]: Can you explain how this is different than line 286?  I don't understand why you're applying this to all of the output.results here.  

As an alternative, could line 286 be the following?
var filesWritten = (options.out)? createFilesWrittenArray(options.out) : createFilesWrittenArray(outputFile);

**Comment by BrandonArp** [comment]
Looks pretty good.  Could you add a test or two to demonstrate the functionality?

File: src/main/resources/typescriptc.js (hunk 1 above)
Comment (DomBlack) [inline src/main/resources/typescriptc.js:301]: The reason for this is that the `createFilesWrittennArray` function calls `fixSourceMapFile`, which is an expensive operation; as it reads the whole source map file in, parses the JSON, maps a small change and then writes it out again.

When operating with the single outFile it is extremely costly to run on a large project (my test project has over 500 source files for instance) as the map file ends up being quite large. This small change saved ~90% of the time for me.

**Comment by wassertim** [comment]
👍  for tests

**Comment by DomBlack** [comment]
@BrandonArp Is this ok to get pulled now or would you like some other changes?

**Comment by BrandonArp** [comment]
Thanks @DomBlack.  I'll get this released right away.

**Comment by BrandonArp** [comment]
Yep. Sorry, I lost track of this after you replied.  I think some refactoring is in order. I'm not particularly happy with the flow of the compiler, but that shouldn't block this PR.

**Pull request merged**